### PR TITLE
[FormLayout] Ensure Item container width is stable

### PR DESCRIPTION
**Pull request by chloerice**
### WHY are these changes introduced?

`FormLayout.Item` currently has an `auto` `flex-basis`, which means when the form initially renders grouped items are equal widths automatically as we expect, but individual items are allowed to grow when their content changes instead of maintaining the equal distribution in the available space. For example, a group of two `TextField` components starts out with the item containers being equal, but an item can expand unexpectedly when a `TextField` dynamically has an error message or if one `TextField` has help text and the other doesn't. 

[CodeSandbox reproduction](https://codesandbox.io/p/sandbox/falling-firefly-xmskmx?file=%2FApp.tsx%3A9%2C1) 

### WHAT is this pull request doing?

This PR sets `flex-basis` to `0`, ensuring sibling `FormLayout.Item`widths are stably equal when grouped.

| Before | After |
|--------|--------|
| Screenshot 2024-05-02 at 4 21 04 PM|Screenshot 2024-05-02 at 4 21 09 PM |

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

🌀 [Spinstance](https://admin.web.form-layout-item-stable-width.chloe-rice.us.spin.dev/store/shop1/customers/new)

### 🎩 checklist

- [ ] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] ~Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)~
- [ ] ~Updated the component's `README.md` with documentation changes~
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/FormLayout/FormLayout.module.css [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .Item {
   /* stylelint-disable-next-line -- Polaris component custom properties */
   --pc-form-layout-item-min-size: 220px;
-  flex: 1 1 auto;
+  flex: 1 1 0;
```
Comment (chloerice) [inline polaris-react/src/components/FormLayout/FormLayout.module.css:4]: (This could just be `flex: 1;`, but I prefer the basis value to be clear)

**Review by sam-b-rose — APPROVED** [review]
Changes look good! 💯 

I had to make the code change manullay on the spin instance to get it to work. I think you just need install the snapshot package and restart Vite to get the flex basis changes to appear. It is still using version `13.2.0`.

| Before | After |
| --- | --- |
| before | after |

**Review by adandach — CHANGES_REQUESTED** [review]
@chloerice , I tophatted using your spin instance by testing the form where we initially discovered the issue. It looks like the issue is persisting. To reproduce you can 
1. enable `internal routes` 
2. go to the [Profile Assessment Dev page](https://admin.web.form-layout-item-stable-width.chloe-rice.us.spin.dev/store/shop1/profile-assessment-dev)
3. Select `DE` (Germany) as the country
4. Select `PERSONAL_DETAILS` as the step
5. Enter nonsense as the postal code


https://github.com/Shopify/polaris/assets/5607510/b848aeae-adfc-4db9-a5dc-bfc3b68995c1

**Review by adandach — APPROVED** [review]
After seeing @sam-b-rose 's comment, I tested in my own spin instance and it looks good! 🚀

**Comment by chloerice** [comment]
Aghhh sorry y'all, thank you @sam-b-rose!

**Pull request merged**